Allow class in `concept` helper.

smathy opened the pull request:
`= concept "foo/bar"` can be: `= concept Foo::Bar` and also `= cell "foo/bar"` can be: `= cell Foo::BarCell`

timoschilling commented:
the same should work with the `cell` helper

smathy commented:
Happy to add it, but let me explain my thoughts first and see if they change your thinking.

First that there's already a difference in the APIs between `cell` and `concept`. Also `concept` already has the newer, less magical API so making it even less magical by allowing the actual class constant seemed much more fitting for `concept` than `cell`. Further to that `cell`s magical API of adding the `"_cell"` suffix would mean that `cell "foo/bar"`would have to become `cell Foo::BarCell` which is internally inconsistent for `cell`'s own API.

timoschilling commented:
I understand what you mean, the reflection from the string argument to the class is different. But if we change only `concept` we have a different API between both. Until now they have both the same args.
Sure there is a difference in the name notation, but I think thats the best way.
```ruby
cell("foo/bar") == cell(Foo::BarCell)
concept("foo/bar") == concept(Foo::Bar)
```

smathy commented:
Ok, so I refactored slightly to take out the duplicate code in both classes, so only the `full_cell_name` method is implemented in both classes, the `camelize.constantize` is only in the base class. I also renamed the `name` parameter to `name_or_class` where appropriate.

timoschilling commented on test/public_test.rb:
```diff
@@ -21,10 +21,13 @@ class Songs < Cell::Concept
   # Concept.cell simply camelizes the string before constantizing.
   it { Cell::Concept.cell("public_test/songs").must_be_instance_of Songs }
 
+  it { Cell::Concept.cell(Songs).must_be_instance_of Songs }
```
I think this should be
```ruby
it { Cell::Concept.cell(PublicTest::Songs).must_be_instance_of Songs }
```

smathy replied:
Yep, better :)

timoschilling commented:
it look's like your are loving the close / open button ;)

timoschilling commented:
good work :+1:

smathy commented:
Yeah, close/open is the only way I can trigger a new Travis run for those occasional failures. Using that `@response.body.must_equal` is a bad idea without a `describe` context.  I'll send a PR to fix that too :)

smathy commented:
Ok, so fixed that test failure in #338, could merge that to master and rebase this to get past those annoying travis false negatives.

smathy commented:
:shipit: baby!  Do you merge stuff @timoschilling - or is that only @apotonick ?

timoschilling commented:
No thats only @apotonick, I'm only a normal user in this repo

smathy commented:
:+1:

smathy commented:
Bump, how about this @apotonick ?

apotonick commented:
Why so complicated?

![screenshot_2016-04-18_21-51-03](https://cloud.githubusercontent.com/assets/10406/14603204/c94332a6-05af-11e6-880a-fbc6f16d9fed.png)


Thanks for that, @smathy ! :heart:

smathy commented:
Did I miss some simpler way to do that?
